### PR TITLE
Enable recursively list dataset files in data_dir subfolders in image_processing annotation converter

### PR DESCRIPTION
**Pull request by mzuevx**

### CODE REVIEW

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

File: image_processing.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
+"""
```
Comment (eaidova) [inline image_processing.py:1]: What is it? Why it should be located in root of the repo???

File: tools/accuracy_checker/accuracy_checker/annotation_converters/image_processing.py [hunk 2]
```diff
@@ -58,22 +62,30 @@ def configure(self):
...
         self.annotation_loader = LOADERS_MAPPING.get(self.get_value_from_config('annotation_loader'))
         if not self.annotation_loader:
             raise ConfigError('provided not existing loader')
 
     def convert(self, check_content=False, progress_callback=None, progress_interval=100, **kwargs):
         content_errors = [] if check_content else None
         file_list_in = []
-        for file_in_dir in self.data_dir.iterdir():
+        if self.recursive:
+            data_dir_files = [file for file in self.data_dir.glob('**/*') if file.is_file()]
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/image_processing.py:74]:
```suggestion
            data_dir_files = [file for file in self.data_dir.rglob('*') if file.is_file()]
```
https://docs.python.org/3/library/pathlib.html#pathlib.Path.rglob

File: tools/accuracy_checker/accuracy_checker/annotation_converters/image_processing.py [hunk 3]
```diff
@@ -58,22 +62,30 @@ def configure(self):
...
                 file_list_in.append(file_in_dir)
 
         annotation = []
         num_iterations = len(file_list_in)
         for in_id, in_file in enumerate(file_list_in):
-            in_file_name = in_file.parts[-1]
-            gt_file_name = self.out_suffix.join(in_file_name.split(self.in_suffix))
+            in_file_name = in_file.relative_to(self.data_dir) if self.recursive else in_file.parts[-1]
+            if self.recursive:
+                gt_file_name = Path(self.out_suffix.join(str(in_file_name).split(self.in_suffix)))
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/image_processing.py:86]: why do you need to wrap into Path?
it is unsafe store path in pickle (Path class is not OS agnostic)

File: tools/accuracy_checker/accuracy_checker/annotation_converters/image_processing.py [hunk 4]
```diff
@@ -58,22 +62,30 @@ def configure(self):
...
+        for file_in_dir in data_dir_files:
             if self.in_suffix in file_in_dir.parts[-1]:
                 file_list_in.append(file_in_dir)
 
         annotation = []
         num_iterations = len(file_list_in)
         for in_id, in_file in enumerate(file_list_in):
-            in_file_name = in_file.parts[-1]
-            gt_file_name = self.out_suffix.join(in_file_name.split(self.in_suffix))
+            in_file_name = in_file.relative_to(self.data_dir) if self.recursive else in_file.parts[-1]
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/image_processing.py:84]:
```suggestion
            in_file_name = str(in_file.relative_to(self.data_dir)) if self.recursive else in_file.parts[-1]
```

File: tools/accuracy_checker/accuracy_checker/annotation_converters/image_processing.py (hunk 3 above)
Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/image_processing.py:86]: I believe, after these suggestion  https://github.com/openvinotoolkit/open_model_zoo/pull/2199/files#r594196221
behaviour for recursive and non-recursive will be the same

File: tools/accuracy_checker/accuracy_checker/annotation_converters/README.md [hunk 5]
```diff
@@ -172,6 +172,7 @@ Accuracy Checker supports following list of annotation converters and specific f
   * `data_dir` - path to folder, where images in low and high resolution are located.
   * `input_suffix` - input file name's suffix (default `in`).
   * `target_suffix` - target ground truth file name's suffix (default `out`).
+  * `recursive` - enables acquiring of dataset files from `data_dir` subcatalogs (default False)
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/README.md:175]:
```suggestion
  * `recursive` - enables acquiring of dataset files from `data_dir` subcatalogs (default False).
```

**Comment by eaidova** [comment]
Jenkins ok to test

**Comment by mzuevx** [comment]
Jenkins please retry a build

**Review by eaidova — APPROVED** [review]

**Comment by eaidova** [comment]
Jenkins please retry a build

**Pull request merged**